### PR TITLE
BSM-51: Implementation of hierarchy layout for topology map

### PR DESCRIPTION
**Pull request by muratyildiz1976**
Implemented an algorithm to transform a layout (e.g. circle layout) to a hierarchy layout.

Jira: http://issues.opennms.org/browse/BSM-51
Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS250

### CODE REVIEW

**Comment by indigo423** [comment]
@muratyildiz1976 This looks better.

**Comment by indigo423** [comment]
:trophy:

**Comment by muratyildiz1976** [comment]
:+1:  i was lucky another colleague from germany was still not sleeping too :)

**Comment by indigo423** [comment]
You're welcome. Now I have Toto sleep ;) gn8

File: features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/jung/HierarchyLayoutAlgorithm.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+ *          <li>{@link org.opennms.features.topology.app.internal.jung.RealUltimateLayoutAlgorithm}
+ *          <li>{@link org.opennms.features.topology.app.internal.jung.SpringLayoutAlgorithm}
+ *          <li>{@link org.opennms.features.topology.app.internal.jung.TopoFRLayoutAlgorithm}
+ *     </ul><p>
+ *
+ * @see <a href="http://jung.sourceforge.net">Java Universal Network/Graph Framework</a>
+ *
+ *
+ */
+public class HierarchyLayoutAlgorithm extends AbstractLayoutAlgorithm {
```
Comment (mvrueden) [inline features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/jung/HierarchyLayoutAlgorithm.java]: Please add a javadoc statement about the assumptions about the provided graph:
* The algorithm generates the layout correctly if only one root element exists or if there are no vertices at all
* If no root element exists, a random vertex from the graph is picked and defined as root.
* If multiple root elements exist, the first one returned by `getDisplayedVertices()` which has no parent (= root candidate) is picked as the root element

**Comment by mvrueden** [comment]
Hey @muratyildiz1976 , this looks good. Please add a couple of more sprinkles to the javadoc as mentioned above. 
If that was done, this can be merged.
Please DO NOT MERGE before https://github.com/OpenNMS/opennms/pull/435. (BSM-50)

**Comment by mvrueden** [comment]
As soon as Bamboo turns green, this can be merged.

**Pull request merged**